### PR TITLE
Version 0.21.0

### PR DESCRIPTION
**Pull request by Kludex**
Hi there 👋 

You want the next Starlette release? Help me to review the checklist. :)

## Checklist

- [x] #1263
- [x] #1649 [APPROVED]
- [x] ~#1701~ postponing to 0.22.0.
- [x] #1376 [APPROVED]
- [x] #1715
- [X] A formatter to migrate from `requests` to `httpx` based TestClient. This is mainly to help our users to migrate. 🙏 Good enough: https://github.com/Kludex/bump-testclient/
- [x] #1863
- [x] #1864

The checklist is to make a decision, not necessarily to accept it. 🙏 

## Info

Compare: https://github.com/encode/starlette/compare/0.20.4...master
Draft: https://github.com/encode/starlette/releases/tag/untagged-19fff6769435226ef06b

### CODE REVIEW

**Comment by alex-oleshkevich** [comment]
That's big!

**Comment by Kludex** [comment]
> That's big!

Should we make a release with what we have?

@tomchristie Can you invite @alex-oleshkevich to encode, please? 🙏

**Comment by alex-oleshkevich** [comment]
Is there any schedule? It would be valuable for 3rd-party to predict when a new release happens. Currently it seems it is pretty random.

**Comment by Kludex** [comment]
> Is there any schedule? It would be valuable for 3rd-party to predict when a new release happens. Currently it seems it is pretty random.

Any member has the power to create a release.

Which kind of predictability are you referring to? Like, is there any package in the Python ecosystem that have this?

**Comment by alex-oleshkevich** [comment]
It could be like the first day of each month if there are enough changes to include. 

Django has 6 month cycle, Ubuntu does so. Starlette while at version 0.* may have more frequent releases.

**Comment by alex-oleshkevich** [comment]
May case: I am hungry to see new test client and middleware in Mount So I can remove hacks from my library. I basically want to see a date when I can do that :)

**Comment by Kludex** [comment]
> It could be like the first day of each month if there are enough changes to include.
> 
> Django has 6 month cycle, Ubuntu does so. Starlette while at version 0.* may have more frequent releases.

I think that's something to consider for when Starlette is on 1.0+.

> May case: I am hungry to see new test client and middleware in Mount So I can remove hacks from my library. I basically want to see a date when I can do that :)

I'll try to speed up. 🥺 🙏

File: docs/release-notes.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+## 0.21.0
+
```
Comment (Kludex) [inline docs/release-notes.md:2]:
```suggestion

September 21, 2022

This release replaces the underneath HTTP client used on the `TestClient` (`requests` :arrow_right: `httpx`), and as those clients [differ _a bit_ on their API](https://www.python-httpx.org/compatibility/), your test suite will likely break. To make the migration smoother, you can use the [`bump-testclient`](https://github.com/Kludex/bump-testclient) tool.

### Changed
* Replace HTTP client on TestClient from `requests` to `httpx` [#1376](https://github.com/encode/starlette/pull/1376).

### Added
* Add `WebSocketException` and support for WS handlers [#1263](https://github.com/encode/starlette/pull/1263).
* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).
```

File: docs/release-notes.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+September 21, 2022
+
+This release replaces the underneath HTTP client used on the `TestClient` (`requests` :arrow_right: `httpx`), and as those clients [differ _a bit_ on their API](https://www.python-httpx.org/compatibility/), your test suite will likely break. To make the migration smoother, you can use the [`bump-testclient`](https://github.com/Kludex/bump-testclient) tool.
+
+### Changed
+* Replace HTTP client on TestClient from `requests` to `httpx` [#1376](https://github.com/encode/starlette/pull/1376).
+
+### Added
+* Add `WebSocketException` and support for WS handlers [#1263](https://github.com/encode/starlette/pull/1263).
+* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).
```
Comment (Kludex) [inline docs/release-notes.md:12]:
```suggestion
* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).

```

File: docs/release-notes.md [hunk 3]
```diff
@@ -1,3 +1,16 @@
+## 0.21.0
+
+September 21, 2022
```
Comment (Kludex) [inline docs/release-notes.md]:
```suggestion
September 24, 2022
```

File: docs/release-notes.md [hunk 4]
```diff
@@ -1,3 +1,16 @@
...
+
+This release replaces the underneath HTTP client used on the `TestClient` (`requests` :arrow_right: `httpx`), and as those clients [differ _a bit_ on their API](https://www.python-httpx.org/compatibility/), your test suite will likely break. To make the migration smoother, you can use the [`bump-testclient`](https://github.com/Kludex/bump-testclient) tool.
+
+### Changed
+* Replace HTTP client on TestClient from `requests` to `httpx` [#1376](https://github.com/encode/starlette/pull/1376).
+
+### Added
+* Add `WebSocketException` and support for WS handlers [#1263](https://github.com/encode/starlette/pull/1263).
+* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).
```
Comment (Kludex) [inline docs/release-notes.md:12]:
```suggestion
* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).
* Implement `__repr__` for route classes [#1864](https://github.com/encode/starlette/pull/1864).
```

File: docs/release-notes.md [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
+September 24, 2022
+
+This release replaces the underneath HTTP client used on the `TestClient` (`requests` :arrow_right: `httpx`), and as those clients [differ _a bit_ on their API](https://www.python-httpx.org/compatibility/), your test suite will likely break. To make the migration smoother, you can use the [`bump-testclient`](https://github.com/Kludex/bump-testclient) tool.
+
+### Changed
+* Replace HTTP client on TestClient from `requests` to `httpx` [#1376](https://github.com/encode/starlette/pull/1376).
+
+### Added
+* Add `WebSocketException` and support for WS handlers [#1263](https://github.com/encode/starlette/pull/1263).
+* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).
```
Comment (Kludex) [inline docs/release-notes.md:12]:
```suggestion
* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).
* Oficially support Python 3.11 [1863](https://github.com/encode/starlette/pull/1863).
```

File: docs/release-notes.md [hunk 6]
```diff
@@ -1,3 +1,18 @@
...
+* Oficially support Python 3.11 [1863](https://github.com/encode/starlette/pull/1863).
+* Implement `__repr__` for route classes [#1864](https://github.com/encode/starlette/pull/1864).
```
Comment (Kludex) [inline docs/release-notes.md:14]:
```suggestion
* Implement `__repr__` for route classes [#1864](https://github.com/encode/starlette/pull/1864).

### Fixed
* Fix bug on which `BackgroundTasks` were cancelled when using `BaseHTTPMiddleware` and client disconnected [#1715](https://github.com/encode/starlette/pull/1715).
```

File: docs/release-notes.md [hunk 7]
```diff
@@ -1,3 +1,21 @@
...
+
+This release replaces the underneath HTTP client used on the `TestClient` (`requests` :arrow_right: `httpx`), and as those clients [differ _a bit_ on their API](https://www.python-httpx.org/compatibility/), your test suite will likely break. To make the migration smoother, you can use the [`bump-testclient`](https://github.com/Kludex/bump-testclient) tool.
+
+### Changed
+* Replace HTTP client on TestClient from `requests` to `httpx` [#1376](https://github.com/encode/starlette/pull/1376).
+
+### Added
+* Add `WebSocketException` and support for WS handlers [#1263](https://github.com/encode/starlette/pull/1263).
+* Add `middleware` parameter to `Mount` class [#1649](https://github.com/encode/starlette/pull/1649).
+* Oficially support Python 3.11 [1863](https://github.com/encode/starlette/pull/1863).
```
Comment (Kludex) [inline docs/release-notes.md]:
```suggestion
* Officially support Python 3.11 [1863](https://github.com/encode/starlette/pull/1863).
```

File: docs/release-notes.md [hunk 8]
```diff
@@ -1,3 +1,21 @@
+## 0.21.0
+
+September 24, 2022
+
+This release replaces the underneath HTTP client used on the `TestClient` (`requests` :arrow_right: `httpx`), and as those clients [differ _a bit_ on their API](https://www.python-httpx.org/compatibility/), your test suite will likely break. To make the migration smoother, you can use the [`bump-testclient`](https://github.com/Kludex/bump-testclient) tool.
```
Comment (adriangb) [inline docs/release-notes.md]:
```suggestion
This release replaces the underlying HTTP client used on the `TestClient` (`requests` :arrow_right: `httpx`), and as those clients [differ _a bit_ on their API](https://www.python-httpx.org/compatibility/), your test suite will likely break. To make the migration smoother, you can use the [`bump-testclient`](https://github.com/Kludex/bump-testclient) tool.
```

File: docs/release-notes.md [hunk 9]
```diff
@@ -1,3 +1,21 @@
...
+
+### Changed
+* Replace HTTP client on TestClient from `requests` to `httpx` [#1376](https://github.com/encode/starlette/pull/1376).
```
Comment (adriangb) [inline docs/release-notes.md]:
```suggestion
* Replace `requests` with `httpx` in `TestClient` [#1376](https://github.com/encode/starlette/pull/1376).
```

File: docs/release-notes.md [hunk 10]
```diff
@@ -1,3 +1,21 @@
...
+
+### Added
+* Add `WebSocketException` and support for WS handlers [#1263](https://github.com/encode/starlette/pull/1263).
```
Comment (adriangb) [inline docs/release-notes.md]:
```suggestion
* Add `WebSocketException` and support for WebSocket error handlers [#1263](https://github.com/encode/starlette/pull/1263).
```

**Review by adriangb — APPROVED** [review]

File: docs/release-notes.md (hunk 10 above)
Comment (Kludex) [reply docs/release-notes.md]: We differentiate between error and exception handlers, but good point 👍 
```suggestion
* Add `WebSocketException` and support for WebSocket exception handlers [#1263](https://github.com/encode/starlette/pull/1263).
```

File: docs/release-notes.md [hunk 11]
```diff
@@ -1,3 +1,21 @@
+## 0.21.0
+
+September 24, 2022
```
Comment (Kludex) [inline docs/release-notes.md]:
```suggestion
September 26, 2022
```

**Pull request merged**